iterable-extension/src/main/java/com/mparticle/ext/iterable/IterableExtension.java

```diff
@@ -237,13 +239,51 @@ void updateUser(EventProcessingRequest request) throws IOException {
             addUserIdentitiesToRequest(userUpdateRequest, request);
             if (!isEmpty(userUpdateRequest.email) || !isEmpty(userUpdateRequest.userId)) {
                 userUpdateRequest.dataFields = convertAttributes(request.getUserAttributes(), shouldCoerceStrings(request));
+                if (request.getUserAttributeLists() != null && !request.getUserAttributeLists().isEmpty()) {
+                    Map<String, List<Object>> processedUserAttributeList = processUserAttributesList(request.getUserAttributeLists(), shouldCoerceStrings(request));
+                    for (Map.Entry<String, List<Object>> entry : processedUserAttributeList.entrySet()) {
+                        String key = entry.getKey();
+                        Object value = entry.getValue();
+                        userUpdateRequest.dataFields.put(key, value);
+                    }
+                }
                 Call<IterableApiResponse> preparedCall = iterableService.userUpdate(getApiKey(request), userUpdateRequest);
                 Response<IterableApiResponse> response = makeIterableRequest(preparedCall, request.getId());
                 handleIterableResponse(preparedCall, response, request.getId());
             }
         }
     }
 
+    /**
+     * Returns a type-corrected userAttributeList
+     *
+     * @param userAttributesList Map of <keys>: List<String values>
+     * @param shouldCoerceStrings boolean setting in mParticle
+     * @return a type-corrected map
+     */
+    private Map<String, List<Object>> processUserAttributesList(Map<String, List<String>> userAttributesList, boolean shouldCoerceStrings) {
+        Map<String, List<Object>> convertedAttributeList = new HashMap<>();
+        userAttributesList.forEach((key, value)-> {
+            value.forEach((attribute)-> {
+                if (!convertedAttributeList.containsKey(key)) {
+                    convertedAttributeList.put(key, new ArrayList<>());
+                }
+                if (isValidJson(attribute)) {
+                    Map<String, Object> typeCorrectedMap = jsonToMap(attribute);
+                    convertedAttributeList.get(key).add(typeCorrectedMap);
```

Comment:
The `isValidJson` method presumably returns `true` if `attribute` contains a JSON array, string, number, etc., not just a JSON object. But then `typeCorrectedMap` will be `null` for the non-object cases; is that acceptable to customers?

Reply:
mParticle only sends us strings, and will stringify an array, object, list of objects, etc. I've updated this code to to avoid the `null` case you pointed out, and cast json lists to a `List`
- json --> map
- json --> list

In all other cases (or cases that fail to coerce to map/list), the code will read in the values as a string. Can you imagine another case where this might fail?